### PR TITLE
Custom miner for anvil

### PR DESCRIPTION
**Pull request by skudasov**
- Sub-second custom miner for anvil
- Some debug methods
- Batch-oriented custom miner to mine huge blocks
- Smoke tests

### CODE REVIEW

File: client/miner_test.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"math/big"
+	"testing"
+	"time"
+)
+
+type AnvilContainer struct {
```
[the remark this answers is not in the document]
Comment (skudasov) [reply client/miner_test.go:19]: We can add it later, here we have it only for testing purposes and we already have anvil for K8s.

File: client/miner_test.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+func TestAnvilMiner(t *testing.T) {
+	t.Run("mine periodically", func(t *testing.T) {
+		period := 500 * time.Millisecond
+		iterations := 10
+		ac, err := startAnvil()
+		require.NoError(t, err)
+		client, err := ethclient.Dial(ac.URL)
+		require.NoError(t, err)
+		pm := NewAnvilMiner(ac.URL)
+		pm.MinePeriodically(500 * time.Millisecond)
```
[the remark this answers is not in the document]
Comment (skudasov) [reply client/miner_test.go:104]: What that would give us? Test asserts that we mine 2 blocks per second and in the end we have "iterations" blocks.

File: client/miner_test.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		pm.MineBatch(txnInBlock, 1*time.Second, 1*time.Minute)
+		time.Sleep(sendTransactionEvery * time.Duration(iterations) * 2)
+		pm.Stop()
+		stopTxns <- struct{}{}
+		bn, err := client.BlockNumber(context.Background())
+		require.NoError(t, err)
+		for i := 1; i <= int(bn); i++ {
+			block, err := client.BlockByNumber(context.Background(), big.NewInt(int64(i)))
+			require.NoError(t, err)
+			require.GreaterOrEqual(t, int64(block.Transactions().Len()), txnInBlock)
```
[the remark this answers is not in the document]
Comment (skudasov) [reply client/miner_test.go:130]: We just need to be sure block contains strictly "batch number" transactions

File: client/miner_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (skudasov) [reply client/miner_test.go:19]: Tbh, in almost all the cases where I can think about `anvil` I prefer to run such tests in K8s where we can do soak/chaos/gasSpikes/etc

**Review by kalverra — APPROVED** [review]

**Pull request merged**